### ISSUE TITLE
Does not heed pre-commits file filters.

### ISSUE THREAD

**Issue by daniel-wong-dfinity-org** [ansible/ansible-lint#3171]
##### Summary

If I do

```yaml
files: DOES_NOT_EXIST
```

in my .pre-commit-config.yaml, ansible-lint still scans my files and says they are wrong, even though no path matches the regular expression `DOES_NOT_EXIST`. (We have many yaml files, and many of them are not ansible.)

Same thing seems to happen if I do this:

```yaml
files: this/is/an/ansible/file.yaml
```

I.e. ansible-lint does not care and just goes ahead and scans all files in my repo.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

I have tried the following values in my .pre-commit-config.yaml:

```yaml
rev: v4.3.0
```

and

```yaml
# Latest as of now.
rev: 6.14.2
```

OS: macOS Monterey 12.6.3 (Intel)

##### Desired Behavior

Only scan files that match pre-commit file filters.

IIUC, pre-commit simply passes all the paths of matching files on the command line, so there should be no need to read the pre-commit-config.yaml file to figure out what files should be considered eligible.

##### Actual Behavior

Scans all files (apparently), many of which are not for ansible.

**Comment by daniel-wong-dfinity-org** [comment]
Ah. This (extremely surprising and IMHO not good) behavior is intentional: https://github.com/ansible/ansible-lint/blob/36ef08b965cb63a8a7d1efafc45cd569e1a4f87f/.pre-commit-hooks.yaml#L11

**Comment by ssbarnea** [comment]
That is indeed on purpose and for good reasons, as running with individual files will prevent some key features from working in ansible-lint and give false positive results.
